libs/core/src/lib/popover/popover-service/popover.service.ts

```diff
@@ -208,6 +208,7 @@ export class PopoverService extends BasePopoverClass {
      */
     refreshConfiguration(config: BasePopoverClass): void {
         const onlyChanged = Object.keys(new BasePopoverClass()).filter((key) => this[key] !== config[key]);
+        this.appendTo = config.appendTo;
```

Comment:
New BasePopoverClass.appendTo will be new input, if new input is different from the one which is in PopoverService. Then on line 225, of this file, that different is going to be assigned to `this`, So I do not see any reason for this specific explicit assignment

Reply:
Exactly, but it is not getting assigned on line 225 which causes the bug.

So I went deeper and found out that when we do `new BasePopoverClass()`, the `appendTo` key is not there, because it is `undefined`. We can see in the screenshot below that `config.appendTo` exists but `basePopoverClass` doesn't have this key, so its never added to the `onlyChanged`:

![image](https://user-images.githubusercontent.com/6796342/231097030-65f25029-6692-456e-b0fa-5a5fd84ca48d.png)

Instead of having the explicit assignment, we could instead assign `null` as a default value to the `BasePopoverClass.appendTo`, so the key will be there. I've tested and it worked as expected, so maybe we could go with this approach instead.

Reply:
That's better, yes

Reply:
cool, its done :)